### PR TITLE
Add documentation for the api and expand README

### PR DESCRIPTION
**Pull request by MilGard91**
## Context

Issue: #72 

## Changes proposed in this pull request

- Add openapi documentation for the routes
- Expand README with missing information

## Test plan

- N/A

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [x] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [ ] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,51 @@
-# cover-router
+# Cover Router
 Computes optimal capacity allocation per Nexus Mutual staking pool on cover purchase
```
Comment (roxdanila) [inline README.md:2]: maybe rephrase to: 
`Computes the optimal route for capacity allocation in order to get the best price on cover purchases.`

File: src/routes/quote.js [hunk 2]
```diff
@@ -7,6 +7,96 @@ const { TARGET_PRICE_DENOMINATOR } = require('../lib/constants');
...
+ *                             type: string
+ *                             description: The pool id
+ *                           coverAmountInAsset:
+ *                             type: string
+ *                             description: The cover amount in asset for the pool
+ *                           skip:
+ *                             type: boolean
+ *                             description: Skip
+ *                             default: false
+ *                 capacities:
```
Comment (mixplore) [inline src/routes/quote.js:80]: We should describe this more so it doesn't create confusion. Saying something like: it's the cheapest capacity allocated from available pools/pools with available capacity based on the requested amount. And maybe mention this is obtained based on the multipool router 🤔 
Trying to avoid the confusion I had thinking we could add up the capacities from
here to get the max capacity - so extra info would help I believe.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,51 @@
...
+## Setup
+
+1. `git clone git@github.com:NexusMutual/cover-router.git`
+2. `cd cover-router && npm ci`
+3. `cp .env.sample .env`
+    1. Set environment variables in .env:
+    2. CONCURRENCY_NUMBER= number of concurrent products to process for faster startup
+    3. PROVIDER_URL - use either [Infura](https://www.infura.io/) or [Alchemy](https://www.alchemy.com/)
+    4. PORT= port number for the server
+    5. Contract addresses are being pull from the NexusMutual deployments repo
```
[the remark this answers is not in the document]
Comment (mixplore) [reply README.md]: 🤔  I'm not sure either, but I left it in there for now - probably worth removing though

File: src/routes/capacity.js [hunk 4]
```diff
@@ -15,6 +15,47 @@ const formatCapacityResult = ({ productId, availableCapacity, usedCapacity, minA
...
+ *                           description: The capacity amount
+ *                   allocatedNxm:
+ *                     type: string
+ *                     description: The allocated NXM
+ *                   minAnnualPrice:
+ *                     type: string
+ *                     description: The minimal annual price
+ *                   maxAnnualPrice:
+ *                     type: string
+ *                     description: The maximal annual price
```
[the remark this answers is not in the document]
Comment (mixplore) [reply src/routes/capacity.js]: 🤔  For the prices in particular, I think `integer` would be more confusing given that they actually represent percentages in the end. Not sure  what the best approach would be, I just added a more detailed description for now.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,51 @@
...
+best available combination of pools for the premium.
+- **Quote Engine**: Calculation is based on batching the current capacities and mapping them with the
+  price per pool, then the cheapest capacity is selected, and remapped if the whole cover amount is not covered by the
+  cheapest capacity.
+
+### Capacity Route
+- **URL**: `v2/api/capacity`
+- **Method**: `GET`
+- **OpenAPI**: `/v2/api/docs/#/Capacity/get_api_v2_capacity_`
+- **Description**: Returns the current capacity for all products for the period of 30 days.
```
Comment (roxdanila) [inline README.md:44]: it has an optional param, we should mention that. the default is 30 days

File: README.md [hunk 6]
```diff
@@ -1,2 +1,51 @@
...
+
+### Capacity Route for a specific product
+- **URL**: `v2/api/capacity/{productId}`
+- **Method**: `GET`
+- **OpenAPI**: `/v2/api/docs/#/Capacity/get_api_v2_capacity__productId_`
+- **Description**: Returns the current capacity for specific product for the period of 30 days.
```
Comment (roxdanila) [inline README.md:50]: same as above

File: README.md (hunk 5 above)
Comment (mixplore) [reply README.md:44]: Would this work: `Returns the current capacity for specific product for the period of 30 days if no period query param is specified.` ?

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (MilGard91) [reply README.md]: I agree

File: README.md (hunk 5 above)
Comment (MilGard91) [reply README.md:44]: Works for me, just make sure here it says for all products and on next one to be for specific product

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mixplore) [reply README.md]: Removed it 🙌

**Review by mixplore — APPROVED** [review]

File: src/routes/quote.js [hunk 7]
```diff
@@ -7,6 +7,101 @@ const { TARGET_PRICE_DENOMINATOR } = require('../lib/constants');
...
+ *               properties:
+ *                 quote:
+ *                   type: object
+ *                   properties:
+ *                     totalCoverAmountInAsset:
+ *                       type: string
+ *                       description: The total cover amount value in cover asset (smallest unit e.g. wei for ETH)
+ *                     annualPrice:
+ *                       type: string
+ *                       description: The annual price as a percentage value (2 decimals). Should be divided by 100.
```
Comment (rackstar) [inline src/routes/quote.js:58]: I believe the `annualPrice` is an integer value between 0-10000. i.e. 750 is 7.5%

File: src/routes/quote.js [hunk 8]
```diff
@@ -7,6 +7,101 @@ const { TARGET_PRICE_DENOMINATOR } = require('../lib/constants');
...
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 quote:
+ *                   type: object
+ *                   properties:
+ *                     totalCoverAmountInAsset:
+ *                       type: string
+ *                       description: The total cover amount value in cover asset (smallest unit e.g. wei for ETH)
```
Comment (rackstar) [inline src/routes/quote.js:55]: for the values that are wei. is it worth adding `format: Integer` to convey a bit better that is a string integer that we are expecting?
